atomic_reactor/plugins/build_imagebuilder.py

```diff
@@ -43,36 +38,28 @@ def run(self):
 
         image = builder.image.to_str()
         # TODO: directly invoke go imagebuilder library in shared object via python module
-        kwargs = dict(stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        kwargs = dict(stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True)
+        encoding_params = dict(encoding='utf-8', errors='replace')
         if not PY2:
-            kwargs['encoding'] = 'utf-8'
+            kwargs.update(encoding_params)
         ib_process = subprocess.Popen(['imagebuilder', '-t', image, builder.df_dir], **kwargs)
 
         self.log.debug('imagebuilder build has begun; waiting for it to finish')
-        (output, last_error) = ([], None)
+        output = []
         while True:
             poll = ib_process.poll()
-            # NOTE: imagebuilder writes both stdout and stderr in normal operation.
-            # Because the two streams are not always logged in the same order as they're
-            # produced, prefix logs with stderr/stdout to distinguish the streams.
-            out = sixdecode(ib_process.stdout.readline())
+            out = ib_process.stdout.readline()
```

Comment:
Wouldn't this hang if there's no output?

Reply:
it's supposed to block until there is output. and if the process ends / stream closes, return ''

it works, most of the time.

Reply:
Nope, it just returns an empty string:
```python
>>> import subprocess
>>> kwargs = dict(stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True)
>>> p = subprocess.Popen(['echo', 'yolo'], **kwargs)
>>> p.poll()
0
>>> p.stdout.readline()
'yolo\n'
>>> p.stdout.readline()
''
>>> p.stdout.readline()
''

```